packages/electron-updater/src/MacUpdater.ts

```diff
@@ -79,12 +80,26 @@ export class MacUpdater extends AppUpdater {
       throw newError(`ZIP file not provided: ${safeStringifyJson(files)}`, "ERR_UPDATER_ZIP_FILE_NOT_FOUND")
     }
 
+    const provider = downloadUpdateOptions.updateInfoAndProvider.provider
+
     return this.executeDownload({
       fileExtension: "zip",
       fileInfo: zipFileInfo,
       downloadUpdateOptions,
-      task: (destinationFile, downloadOptions) => {
-        return this.httpExecutor.download(zipFileInfo.url, destinationFile, downloadOptions)
+      task: async (destinationFile, downloadOptions) => {
+        const currentVersionFileName = "currentVersion"
+        const currentVersion = readFileSync(currentVersionFileName, "utf8")
```

Comment:
```
        const currentVersionFileName = "currentVersion"
        const currentVersion = readFileSync(currentVersionFileName, "utf8")
```

Won't this always fail in attempting to read the file? I'm not sure what file we're attempting to read here

Reply:
Also, I think in order to unit test this, we can simply un-skip this test:
```
test.skip("dmg", async () => {
  const outDirs: Array<string> = []
  const tmpDir = new TmpDir("differential-updater-test")
  if (process.env.__SKIP_BUILD == null) {
    await doBuild(outDirs, Platform.MAC.createTarget(undefined, Arch.x64), tmpDir, {
      mac: {
        electronUpdaterCompatibility: ">=2.17.0",
      },
    })
  } else {
    // todo
  }


  await testBlockMap(outDirs[0], path.join(outDirs[1]), MacUpdater, "mac/Test App ßW.app", Platform.MAC)
})
```
https://github.com/electron-userland/electron-builder/blob/master/test/src/updater/differentialUpdateTest.ts#L175-L189

That might do the trick as it should already be generating a dmg and zip? If not, then you'll need to modify it accordingly to get the test up and running.

I haven't worked in this area of the code before but happy to help as much as possible to get this implemented and merged! 🙂